upt_macports/upt_macports.py

```diff
@@ -173,9 +178,38 @@ def create_package(self, upt_pkg, output=None):
         }
 
         try:
+            self.frontend = upt_pkg.frontend
             pkg_cls = pkg_classes[upt_pkg.frontend]
         except KeyError:
             raise upt.UnhandledFrontendError(self.name, upt_pkg.frontend)
-
         packager = pkg_cls()
         packager.create_package(upt_pkg, output)
+
+    def package_version(self, name):
+        pkg_classes = {
+            'pypi': MacPortsPythonPackage,
+            'cpan': MacPortsPerlPackage,
+            'rubygems': MacPortsRubyPackage,
+            'npm': MacPortsNpmPackage
+        }
+        try:
+            port_name = pkg_classes[self.frontend](
+            )._normalized_macports_name(name)
+        except KeyError:
+            raise upt.UnhandledFrontendError(self.name, self.upt_pkg.frontend)
+
+        self.logger.info(f'Checking MacPorts tree for port {port_name}')
+        cmd = f'port info --version {port_name}'
+        port = subprocess.getoutput(cmd)
+        # TODO: version not found when macports tree is not updated
+        if port.startswith('Error'):
+            self.logger.info(f'{port_name} not found in MacPorts tree')
+            return ''
+        elif port.startswith('version'):
+            curr_ver = port.split()[1]
+            self.logger.info(
+                f'Current MacPorts Version for {port_name} is {curr_ver}')
+            return curr_ver
+        else:
+            sys.exit('Macports tree version detection failed, try running '
+                     'port selfupdate')
```

Comment:
If I understand it correctly we only end up in this ```else``` block if the ```port``` command failed. So the actual output of the subprocess command is along the lines of ```'/bin/sh: port: command not found'```. I am fine with catching this as with the ```else``` statement, but then the message that will be printed is weird. Shoudn't is just be something like 
```
sys.exit(f'The command "{cmd}" failed. Pleae make sure you have MacPorts installed and/or your PATH is set-up correctly.')
```

Reply:
Yeah I was supposed to change that message after dealing with the TODO, its updated now.